packages/middleware/src/esbuild-plugins.ts

```diff
@@ -0,0 +1,58 @@
+import path from 'path';
+import * as esbuild from 'esbuild';
+
+const processInjectFile = `
+const env = {};
+
+// envOverride is passed by esbuild plugin
+if (typeof envOverride === 'object') {
```

Comment:
Lets perform this check prior to injecting the file

Reply:
Fixed, another good reason to remove that check is that this way the process variable gets tree shaken by esbuild when not accessed